docs/08-developer-guide/01-build/03-docker-images.md

```diff
@@ -2,10 +2,126 @@
 sidebar_label: Docker Images
 ---
 
+<!-- cspell:word testkrb5 -->
+
 # Building Ozone Docker Images
 
-**TODO:** File a subtask under [HDDS-9861](https://issues.apache.org/jira/browse/HDDS-9861) and complete this page or section.
+This page provides an overview of the Docker images maintained by the Apache Ozone community for developing and testing Ozone.  It also describes the workflow to be followed when making changes to one of these images.
+
+## Images
+
+### Base Image (without any Ozone)
+
+[ozone-runner](https://github.com/apache/ozone-docker-runner) contains tools for running and testing Ozone, but does not include any Ozone artifacts.
+
+Developers and CI workflows rely on it heavily to run/test custom Ozone builds (using the local build via bind-mount).  It also serves as the base image for `apache/ozone` (see next section).
+
+Published to [Docker Hub](https://hub.docker.com/r/apache/ozone-runner) and [GitHub](https://github.com/apache/ozone-docker-runner/pkgs/container/ozone-runner).
+
+### Image with Specific Version of Ozone
+
+[ozone](https://github.com/apache/ozone-docker) is built on top of `ozone-runner`, adding the binaries built for official Ozone releases.
+
+These are used for testing compatibility of various Ozone versions, and upgrade from one version to another.  May also be useful for running quick experiments with specific version of Ozone, without the need to download or rebuild it.
+
+Published to [Docker Hub](https://hub.docker.com/r/apache/ozone) and [GitHub](https://github.com/apache/ozone-docker/pkgs/container/ozone).
+
+### Kerberos KDC for Testing
+
+[ozone-testkrb5](https://github.com/apache/ozone-docker-testkrb5) is used as KDC in tests where Kerberos is enabled.
+
+Published only to [GitHub](https://github.com/apache/ozone-docker-testkrb5/pkgs/container/ozone-testkrb5).
```

Comment:
We should explain why this one is not available on docker hub. I think it's because we only use it in our CI?